### PR TITLE
Showing/Hiding OverlayView Fixes

### PR DESCRIPTION
**Pull request by evancrow**
Fixed an issue where the OverlaySheets wouldn't animate to hide.

Also fixed an issue where a different overlay wouldn't show from another overlay. This was caused by a lot of extra calls to `hideOverlay()`. (Also, existing tests should work for this?)

### How was this tested?
- [x] I tested this manually
- [ ] I added automated tests
- [x] Existing automated tests are enough to cover my changes

## Issues addressed
Fixed: #3420

### CODE REVIEW

File: Client/Frontend/Overlay/Sheet/OverlaySheetView.swift [hunk 1]
```diff
@@ -262,7 +262,12 @@ struct OverlaySheetView<Content: View, HeaderContent: View>: View, KeyboardReada
         GeometryReader { outerGeometry in
             DismissBackgroundView(
                 opacity: model.backdropOpacity, position: model.position,
-                onDismiss: style.nonDismissible ? {} : onDismiss)
+                onDismiss: style.nonDismissible
+                    ? {}
+                    : {
+                        onDismiss()
+                        model.overlayManager.hideCurrentOverlay()
```
Comment (darinwf) [inline Client/Frontend/Overlay/Sheet/OverlaySheetView.swift:269]: I wonder if `hideCurrentOverlay` is always risky? Maybe we should have a way to refer to the specific overlay instance we want to hide?

**Review by darinwf — APPROVED** [review]

File: Client/Frontend/Overlay/Sheet/OverlaySheetView.swift (hunk 1 above)
Comment (evancrow) [reply Client/Frontend/Overlay/Sheet/OverlaySheetView.swift:269]: Yes? The problem is that the only instance of the overlay is `currentOverlay` which would always match what is visible, and hide that.

Comment (darinwf) [reply Client/Frontend/Overlay/Sheet/OverlaySheetView.swift:269]: Right, and I assume the issue that we are trying to fix is that the next overlay has become `currentOverlay` but the code is still trying to hide the previous overlay, and the only tool it has to do so is calling `hideCurrentOverlay`. The more robust solution then would be to let code refer to a specific overlay and for their to be no concept of "current overlay" that code can ask to hide.

**Review by darinwf — APPROVED** [review]
Nicely done with the latest commit!

**Pull request merged**